tests/test_check_parallel.py

```diff
@@ -66,20 +66,97 @@ def process_module(self, _astroid):
         self.data.append(record)
 
 
+class ParallelTestChecker(BaseChecker):
+    """A checker that does need to consolidate data.
+
+    To simulate the need to consolidate data, this file only report a
+    message for pairs of files. So if files are treated by threads,
+    some messages will be raised at consolidation time.
+
+    When ran in parallel, a Checker is only given one file at at time:
+    - open
+    - check a single file
+    - close
+
+    (And stats generated by consolidated data) across run invocations.)
+
+    """
+
+    __implements__ = (pylint.interfaces.IRawChecker,)
+
+    name = "parallel-checker"
+    test_data = "parallel"
+    msgs = {
+        "R9999": (
+            "Test %s",
+            "parallel-test-check",
+            "Some helpful text.",
+        )
+    }
+
+    def __init__(self, linter, *args, **kwargs):
+        super().__init__(linter, *args, **kwargs)
+        self.data = []
+        self.linter = linter
+        self.stats = None
+
+    def open(self):
+        """init the checkers: reset statistics information"""
+        self.stats = self.linter.add_stats()
+        self.data = []
+
+    def close(self):
+        for _ in self.data[1::2]:
+            self.add_message("R9999", args=("From process_module, two files seen.",))
```

Comment:
I don't quite get what this simulate, why do we need the ``[1::2]`` ? What is the link with the same construct in reduce_map_data ?

Reply:
Idea is to simulate a checker that has to know multiple files to work, see comment on the class:

> To simulate the need to consolidate data, this file only report a
> message for pairs of files.

This allow me to verify that merging 2 checkers working on 3 files give the same as 3 checkers working on 3 files and 1 checkers on 3 files, ...

I'm not fluent in pylint internals, I struggled to prodce something here, and it revealed multiple things I was not understanding properly when I first pushed on this PR...

I'm adding a comment.

Reply:
I also rephrased the class docstring to try to explain better.

Reply:
Thank you for the explanation and for adding a comment !

Reply:
why start at 1?